### ISSUE TITLE
Sequence not starting

### ISSUE THREAD

**Issue by klassm** [rgc99/irrigation_unlimited#59]
First of all - thanks again for creating this integration! Today I wanted to continue to try out whether it works with my setup. So I took my config, started a run, and .... nothing happened.
I then thought that maybe just the manual run is broken, so I configured a time schedule 2 minutes in the future and waited. Nothing. No log, no water, no starting of anything.
To me it looks like the scheduler is not running, but is there a way I can debug this?

## Version of the custom_component

2022.3.0 (via HACS)

## Configuration

```yaml

controllers:
  zones:
    - name: "Blumenbeet"
      entity_id: "switch.garten_bewaesserung_blumenbeete_safe"
      zone_id: "zone_blumenbeet"
    - name: "Rasen"
      entity_id: "switch.garten_bewaesserung_rasen_safe"
      zone_id: "zone_rasen"
    - name: "Topfplanzen"
      entity_id: "switch.garten_bewaesserung_topfplanzen_safe"
      zone_id: "zone_topfpflanzen"
    - name: "Zaun"
      entity_id: "switch.garten_bewaesserung_zaun_safe"
      zone_id: "zone_zaun"
  sequences:
    - delay: "00:01"
      repeat: 2
      schedules:
        - time: "21:00"
          month: [may, jun, jul, aug, sep]

        - time: "11:48"
      zones:
        - zone_id: "zone_zaun"
          duration: "00:02"
        - zone_id: "zone_blumenbeet"
          duration: "00:05"
        - zone_id: "zone_topfpflanzen"
          duration: "00:02"
        - zone_id: "zone_rasen"
          duration: "00:10"


```

## Debug log

```text

2022-04-23 11:45:47 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-04-23 11:45:00] service: adjust_time, controller: 1, zone: 0, data: {"entity_id": "binary_sensor.irrigation_unlimited_c1_m", "sequence_id": 1, "percentage": 100.0}
2022-04-23 11:45:57 DEBUG (MainThread) [custom_components.irrigation_unlimited] LOAD
2022-04-23 11:45:57 DEBUG (MainThread) [custom_components.irrigation_unlimited] STOP
2022-04-23 11:45:57 DEBUG (MainThread) [custom_components.irrigation_unlimited] START
2022-04-23 11:45:59 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-04-23 11:45:00] service: adjust_time, controller: 1, zone: 0, data: {"entity_id": "binary_sensor.irrigation_unlimited_c1_m", "sequence_id": 1, "percentage": 46.0}
2022-04-23 11:46:01 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-04-23 11:46:00] service: manual_run, controller: 1, zone: 1, data: {"entity_id": ["binary_sensor.irrigation_unlimited_c1_z1"], "time": "0:02:00"}
2022-04-23 11:46:04 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-04-23 11:46:00] service: manual_run, controller: 1, zone: 0, data: {"entity_id": ["binary_sensor.irrigation_unlimited_c1_m"], "time": "0:02:00", "sequence_id": 1}
2022-04-23 11:46:27 DEBUG (MainThread) [custom_components.irrigation_unlimited] INITIALISED
2022-04-23 11:46:27 WARNING (MainThread) [homeassistant.helpers.frame] Detected integration that accesses the database without the database executor; Use homeassistant.components.recorder.get_instance(hass).async_add_executor_job() for faster database operations. Please report issue to the custom component author for irrigation_unlimited using this method at custom_components/irrigation_unlimited/history.py, line 142: data = history.get_significant_states(
2022-04-23 11:47:04 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-04-23 11:47:00] service: disable, controller: 1, zone: 2, data: {"entity_id": ["binary_sensor.irrigation_unlimited_c1_z2"]}
2022-04-23 11:48:18 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-04-23 11:48:00] service: manual_run, controller: 1, zone: 0, data: {"entity_id": ["binary_sensor.irrigation_unlimited_c1_m"], "time": "0:02:00", "sequence_id": 1}

```

After restarting the server I tried to to a manual run, nothing happened. Then I would expect an automatic run. Also nothing. In the meantime I noted that Rasen was on, so I disabled it (actually it was disabled before the restart, but it enabled itself).

Is there anything I can do to debug this further?

**Comment by klassm** [comment]
Out of pure desperation I now also reinstalled the custom component - no effect, behaves the same way.

From the logs during initialization of Home Assistant I also find the init logs for irrigation unlimited:

```
2022-04-23 13:37:47 INFO (MainThread) [custom_components.irrigation_unlimited]
-------------------------------------------------------------------
Irrigation Unlimited
Version: 2022.3.0
If you have any issues with this you need to open an issue here:
https://github.com/rgc99/irrigation_unlimited/issues
-------------------------------------------------------------------

2022-04-23 13:37:47 DEBUG (MainThread) [custom_components.irrigation_unlimited] LOAD
2022-04-23 13:37:48 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-04-23 13:37:00] controller: 0, zone: 0, entity: irrigation_unlimited.coordinator
2022-04-23 13:37:48 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-04-23 13:37:00] service: adjust_time, controller: 1, zone: 0, data: {"percentage": 47.0, "sequence_id": 1}
2022-04-23 13:37:48 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-04-23 13:37:00] service: disable, controller: 1, zone: 2, data: {}
2022-04-23 13:37:48 DEBUG (MainThread) [custom_components.irrigation_unlimited] START
2022-04-23 13:37:49 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-04-23 13:37:00] controller: 1, zone: 0, entity: binary_sensor.irrigation_unlimited_c1_m
2022-04-23 13:37:49 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-04-23 13:37:00] controller: 1, zone: 1, entity: binary_sensor.irrigation_unlimited_c1_z1
2022-04-23 13:37:49 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-04-23 13:37:00] controller: 1, zone: 2, entity: binary_sensor.irrigation_unlimited_c1_z2
2022-04-23 13:37:49 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-04-23 13:37:00] controller: 1, zone: 3, entity: binary_sensor.irrigation_unlimited_c1_z3
2022-04-23 13:37:49 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-04-23 13:37:00] controller: 1, zone: 4, entity: binary_sensor.irrigation_unlimited_c1_z4
```

To my naive eyes that looks like how it should be - except I still cannot schedule anything.

**Comment by klassm** [comment]
Looking at the state attributes there's more information. The controller states that there is no zone:
<img width="349" alt="image" src="https://user-images.githubusercontent.com/498976/164893083-0b4cf3ec-035f-4edb-948e-ebe6237b9a5a.png">

The zone itself is also reporting it will not be scheduled any time soon:
<img width="368" alt="image" src="https://user-images.githubusercontent.com/498976/164893101-89f31c72-b014-4979-bcd9-4602a4eb80bf.png">

**Comment by klassm** [comment]
By fiddling around with the config I now ended up with:
```
controllers:
  - name: "Blumen"
    zones:
      - name: "Blumenbeet"
        entity_id: "switch.garten_bewaesserung_blumenbeete_safe"
      - name: "Topfplanzen"
        entity_id: "switch.garten_bewaesserung_topfplanzen_safe"
      - name: "Zaun"
        entity_id: "switch.garten_bewaesserung_zaun_safe"
    sequences:
      - schedules:
          - time: "21:48"
            month: [apr, may, jun, jul, aug, sep]
        repeat: 2
        zones:
          - zone_id: 1
            duration: "00:01"
          - zone_id: 2
            duration: "00:01"
          - zone_id: 3
            duration: "00:01"
```

At least normal scheduling seems to work now - manual scheduling is still broken -.-

**Comment by rgc99** [comment]
Your configurations do work, let me explain. The `zone_id` parameter in the `controllers:` -> `zones:` section was just recently fixed (now available as a pre-release 2022.4.0). The parameter was being ignored and left with its default value of '1' '2' '3' etc. This means the references under `sequences:` -> `zones:` did not link to the zones as you expect. That's why your last configuration works.

The `time:` in your manual run is too short. When running a sequence this value is the **total** time. Try a value of 6 minutes in your last configuration to run each zone for 1 minute each.

**Comment by klassm** [comment]
Hey @rgc99 ,

thank you so much :-). Now my manual run is also working. Concerning the Ids I will just wait for the new version.

Enjoy your sunday,
Matthias
